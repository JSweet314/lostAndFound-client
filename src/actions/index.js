export const captureUser = user => ({
  type: 'CAPTURE_USER',
  user
});

export const logOutUser = () => ({
  type: 'LOG_OUT_USER'
});

export const submitNewUser = user => ({
  type: 'SUBMIT_NEW_USER',
  user
});

export const captureErrorMessage = message => ({
  type: 'CAPTURE_ERROR_MESSAGE',
  message
});

export const clearErrorMessage = () => ({
  type: 'CLEAR_ERROR_MESSAGE'
});