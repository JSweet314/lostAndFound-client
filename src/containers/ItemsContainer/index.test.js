import React from 'react';
import {shallow} from 'enzyme';
import { ItemsContainer, mapDispatchToProps, mapStateToProps } 
  from '../ItemsContainer';
import * as actions from '../../actions';

describe('ItemsContainer', () => {
  let wrapper;
  const mockItem = {
    itemId: 1,
    locationId: 2,
    status: 'lost',
    name: 'keys',
    description: 'subaru keys',
    date: '02-02-2018'
  };
  const mockGetLocationDetails = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ItemsContainer
        getLocationDetails={mockGetLocationDetails} 
        items={[mockItem]}
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a handleOnClick method', () => {
    wrapper.instance().handleOnClick(1);
    expect(mockGetLocationDetails)
      .toHaveBeenCalledWith({locationId: 2, itemId: 1});
  });

  describe('mapStateToProps', () => {
    it('should return items from the store', () => {
      expect(mapStateToProps({items: []})).toEqual({items: []});
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    it('should call dispatch with action fetchItemLocation', () => {
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.getLocationDetails({});
      expect(mockDispatch)
        .toHaveBeenCalledWith(actions.fetchItemLocation({}));
    });
  });
});

