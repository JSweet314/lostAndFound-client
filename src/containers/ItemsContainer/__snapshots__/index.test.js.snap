// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsContainer should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemsContainer
    getLocationDetails={[Function]}
    items={
        Array [
            Object {
              "date": "02-02-2018",
              "description": "subaru keys",
              "itemId": 1,
              "locationId": 2,
              "name": "keys",
              "status": "lost",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item-map"
>
          <Connect(MapContainer)
                    captureMarkerCoords={[Function]}
                    height="87%"
                    width="75%"
          />
</div>,
        <ul
          className="items-list"
>
          <li
                    className="list-item"
          >
                    <Unknown
                              date="02-02-2018"
                              description="subaru keys"
                              handleOnClick={[Function]}
                              itemId={1}
                              locationId={2}
                              name="keys"
                              status="lost"
                    />
          </li>
</ul>,
      ],
      "className": "items-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(MapContainer)
            captureMarkerCoords={[Function]}
            height="87%"
            width="75%"
/>,
          "className": "item-map",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "captureMarkerCoords": [Function],
            "height": "87%",
            "width": "75%",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="list-item"
>
              <Unknown
                            date="02-02-2018"
                            description="subaru keys"
                            handleOnClick={[Function]}
                            itemId={1}
                            locationId={2}
                            name="keys"
                            status="lost"
              />
</li>,
          ],
          "className": "items-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Unknown
                date="02-02-2018"
                description="subaru keys"
                handleOnClick={[Function]}
                itemId={1}
                locationId={2}
                name="keys"
                status="lost"
/>,
              "className": "list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "date": "02-02-2018",
                "description": "subaru keys",
                "handleOnClick": [Function],
                "itemId": 1,
                "locationId": 2,
                "name": "keys",
                "status": "lost",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item-map"
>
            <Connect(MapContainer)
                        captureMarkerCoords={[Function]}
                        height="87%"
                        width="75%"
            />
</div>,
          <ul
            className="items-list"
>
            <li
                        className="list-item"
            >
                        <Unknown
                                    date="02-02-2018"
                                    description="subaru keys"
                                    handleOnClick={[Function]}
                                    itemId={1}
                                    locationId={2}
                                    name="keys"
                                    status="lost"
                        />
            </li>
</ul>,
        ],
        "className": "items-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(MapContainer)
              captureMarkerCoords={[Function]}
              height="87%"
              width="75%"
/>,
            "className": "item-map",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "captureMarkerCoords": [Function],
              "height": "87%",
              "width": "75%",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="list-item"
>
                <Unknown
                                date="02-02-2018"
                                description="subaru keys"
                                handleOnClick={[Function]}
                                itemId={1}
                                locationId={2}
                                name="keys"
                                status="lost"
                />
</li>,
            ],
            "className": "items-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Unknown
                  date="02-02-2018"
                  description="subaru keys"
                  handleOnClick={[Function]}
                  itemId={1}
                  locationId={2}
                  name="keys"
                  status="lost"
/>,
                "className": "list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "date": "02-02-2018",
                  "description": "subaru keys",
                  "handleOnClick": [Function],
                  "itemId": 1,
                  "locationId": 2,
                  "name": "keys",
                  "status": "lost",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
